server/src/main/java/org/apache/druid/client/BrokerServerView.java

```diff
@@ -112,17 +112,21 @@ public BrokerServerView(
     this.timelines = new HashMap<>();
 
     this.segmentFilter = (Pair<DruidServerMetadata, DataSegment> metadataAndSegment) -> {
+      // Include only watched tiers if specified
       if (segmentWatcherConfig.getWatchedTiers() != null
           && !segmentWatcherConfig.getWatchedTiers().contains(metadataAndSegment.lhs.getTier())) {
         return false;
       }
 
+      // Include only watched datasources if specified
       if (segmentWatcherConfig.getWatchedDataSources() != null
           && !segmentWatcherConfig.getWatchedDataSources().contains(metadataAndSegment.rhs.getDataSource())) {
         return false;
       }
 
-      return true;
+      // Include realtime nodes only if they are watched
+      return segmentWatcherConfig.isWatchRealtimeNodes()
+          || metadataAndSegment.lhs.getType() != ServerType.REALTIME;
```

Comment:
This should be looking for `INDEXER_EXECUTOR`, not sure `REALTIME` is actually used anymore (it was for old realtime nodes I think)

Reply:
yeah, looks like usages were removed in #7915 and just exists now for ... reasons.

Reply:
Thanks a lot for catching this! I can check for `INDEXER_EXECUTOR` here.

Reply:
Considering that we are looking for `INDEXER_EXECUTOR` here, the nomenclature `isWatchRealtimeNodes` is also a little misleading now.

Reply:
we should also add a warning in the code. does calling `isSegmentReplicationTarget()` makes more sense than checking the server type?

Reply:
>does calling isSegmentReplicationTarget() makes more sense than checking the server type?

Hmm, that one is kinda weird too because of `ServerType.BRIDGE` which uh.. predates me and Druid i think, so I'm not really sure its story.

Reply:
maybe `watchRealtimeTasks` would be clearer?

Reply:
`watchRealtimeTasks` works too.